tools/cldr-code/src/main/java/org/unicode/cldr/tool/ChartGrammaticalForms.java

```diff
@@ -455,7 +460,7 @@ public void writeSubcharts(Anchors anchors) throws IOException {
                     final String shortUnit = uc.getShortId(longUnit);
                     String gender = UnitPathType.gender.getTrans(cldrFile, "long", shortUnit, null, null, null, null);
                     final BestUnitForGender bestUnit = unitToBestUnit.get(shortUnit);
-                    if (bestUnit != null) {
+                    if (gender != null && bestUnit != null) {
                         bestUnitForGender.put(gender, bestUnit);
                     }
```

Comment:
same, should just be `(bestUnit != null)` ?

Reply:
ditto (these were from before there was a null check in the code.